controller/registry/dockerclient.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+"""Support the Deis workflow by manipulating and publishing Docker images."""
+
+from __future__ import unicode_literals
+import io
+import logging
+
+from django.conf import settings
+from rest_framework.exceptions import PermissionDenied
+from simpleflock import SimpleFlock
+import docker
+
+logger = logging.getLogger(__name__)
+
+
+class DockerClient(object):
+    """Use the Docker API to pull, tag, build, and push images to deis-registry."""
+
+    FLOCKFILE = '/tmp/controller-pull'
```

Comment:
deis-controller runs `gunicorn` as the "deis" user, so /var/lock isn't writable.